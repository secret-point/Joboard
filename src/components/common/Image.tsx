import { isEmpty } from "lodash";
import React from "react";
import { Text } from "@amzn/stencil-react-components/text";

interface ImageProps {
  src?: string;
  overlayText?: string;
  isContainsOverlay?: boolean;
  imgStyles?: any;
  id?: string;
};

const Image = (props: ImageProps) => {
  const { src, isContainsOverlay, imgStyles, overlayText = "", id } = props;

  return (
    <div id={id} data-testid="image" className="image-container">
      {!isEmpty(src) && (
        <img
          data-testid={`image-${id}`}
          src={src}
          {...imgStyles}
          alt={`image-${id}`}
          aria-hidden="true"
          role="presentation"
          tabIndex={-1}
        />
      )}
      {!isEmpty(src) && isContainsOverlay && (
        <div className="overlay">
          <div className="content">
            <Text>{overlayText}</Text>
          </div>
        </div>
      )}
    </div>
  );
};

export default Image;
