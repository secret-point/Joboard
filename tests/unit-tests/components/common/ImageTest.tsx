import React from "react";
import { shallow } from "enzyme";
import Image from "../../../../src/components/common/Image";

describe("Image", () => {
    it("should match snapshot", () => {
        const shallowWrapper = shallow(
          <Image
            src="https://hiring.amazon.com/#/"
            isContainsOverlay={true}
            overlayText="Test Overlay Text"
            id="IMG001"
            imgStyles={{width: 200, height: 200}}
          />);
    
        expect(shallowWrapper).toMatchSnapshot();
    });
});