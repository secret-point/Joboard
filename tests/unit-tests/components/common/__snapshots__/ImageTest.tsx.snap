// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should match snapshot 1`] = `
<div
  className="image-container"
  data-testid="image"
  id="IMG001"
>
  <img
    alt="image-IMG001"
    aria-hidden="true"
    data-testid="image-IMG001"
    height={200}
    role="presentation"
    src="https://hiring.amazon.com/#/"
    tabIndex={-1}
    width={200}
  />
  <div
    className="overlay"
  >
    <div
      className="content"
    >
      <Text
        font="primary"
        fontSize="T300"
        hyphens="none"
      >
        Test Overlay Text
      </Text>
    </div>
  </div>
</div>
`;
